### PR TITLE
Bump :lang haskell

### PR DESCRIPTION
**Pull request by bixuanzju**
Switch to haskell-language-server, see https://neilmitchell.blogspot.com/2020/09/dont-use-ghcide-anymore-directly.html

### CODE REVIEW

File: modules/lang/haskell/+lsp.el [hunk 1]
```diff
@@ -4,10 +4,8 @@
   :after lsp-mode
   :preface (add-hook 'haskell-mode-local-vars-hook #'lsp!)
   :config
-  (when IS-MAC
```
Comment (bixuanzju) [inline modules/lang/haskell/+lsp.el:7]: I’m using Mac, and I don’t need to do anything here; it works out of the box!

File: modules/lang/haskell/README.org [hunk 2]
```diff
@@ -81,27 +81,15 @@ sudo pacman -S cabal-install ghc
...
-make
+ghcup install hls
 #+END_SRC
 
-*** Arch Linux
-=haskell-ide-engine-git= is available on the AUR
-
-#+BEGIN_SRC bash
-yay -S haskell-ide-engine-git
-#+END_SRC
```
[the remark this answers is not in the document]
Comment (bixuanzju) [reply modules/lang/haskell/README.org:104]: Well, updated platform dependent installation instructions can always be found in the link above. I don’t find it too much useful to put it here.

**Comment by hlissner** [comment]
Would you mind bumping the other packages in this module (`M-x doom/bump-module RET :lang haskell`), then rewording the commit to:

```
Bump :lang haskell

emacs-lsp/lsp-haskell@17d7d4c -> emacs-lsp/lsp-haskell@a56667b
haskell/haskell-mode@41683c0 -> haskell/haskell-mode@e726776
jyp/attrap@4cf3e4a -> jyp/attrap@9c88154
jyp/dante@c516bc9 -> jyp/dante@e2acbf6

Switch to haskell-language-server, see
https://neilmitchell.blogspot.com/2020/09/dont-use-ghcide-anymore-directly.html
```

Bump commits must follow a particular format, so they are machine-readable by future CI.

**Review by hlissner — CHANGES_REQUESTED** [review]
Would you mind bumping the other packages in this module (`M-x doom/bump-module RET :lang haskell`), then rewording the commit to:

```
Bump :lang haskell

emacs-lsp/lsp-haskell@17d7d4c -> emacs-lsp/lsp-haskell@a56667b
haskell/haskell-mode@41683c0 -> haskell/haskell-mode@e726776
jyp/attrap@4cf3e4a -> jyp/attrap@9c88154
jyp/dante@c516bc9 -> jyp/dante@e2acbf6

Switch to haskell-language-server, see
https://neilmitchell.blogspot.com/2020/09/dont-use-ghcide-anymore-directly.html
```

Bump commits must follow a particular format, so they are machine-readable by future CI.

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Thanks for your help!

**Pull request merged**